src/control/lib/hardware/pci.go

```diff
@@ -366,3 +389,109 @@ func NewPCIAddressSet(addrs ...string) (*PCIAddressSet, error) {
 func NewPCIAddressSetFromString(addrs string) (*PCIAddressSet, error) {
 	return NewPCIAddressSet(strings.Split(addrs, bdevPciAddrSep)...)
 }
+
+type (
+	// PCIDevice represents an individual hardware device.
+	PCIDevice struct {
+		Name      string     `json:"name"`
+		Type      DeviceType `json:"type"`
+		NUMANode  *NUMANode  `json:"-"`
+		Bus       *PCIBus    `json:"-"`
+		PCIAddr   PCIAddress `json:"pci_address"`
+		LinkSpeed float64    `json:"link_speed"`
```

Comment:
why do we have a link speed but not width? I would have imagined they would be stored together

Reply:
The hwloc info does not provide a link width value as far as I can tell. I believe that we'll need to incorporate support for a pcilib provider in order to obtain that detail.